lib/3scale/backend/stats/partition_eraser_job.rb

```diff
@@ -0,0 +1,59 @@
+module ThreeScale
+  module Backend
+    module Stats
+      # Job for deleting service stats
+      # Perform actual key deletion from a key partition definition
+      class PartitionEraserJob < BackgroundJob
+        # low priority queue
+        @queue = :stats
+
+        class << self
+          include StorageHelpers
+          include Configurable
+
+          def perform_logged(_enqueue_time, service_id, applications, metrics, users,
+                             from, to, offset, length, context_info = {})
+            job = DeleteJobDef.new(
+              service_id: service_id,
+              applications: applications,
+              metrics: metrics,
+              users: users,
+              from: from,
+              to: to
+            )
+
+            validate_job(job, offset, length)
+
+            stats_key_gen = KeyGenerator.new(job.to_hash)
+
+            stats_key_gen.keys.drop(offset).take(length).each_slice(configuration.stats.delete_batch_size) do |slice|
```

Comment:
I'm still not convinced about the idea of generating the whole set of keys on every `PartitionEraserJob`. It'd be good to hear other opinions on this. @unleashed ? @miguelsorianod ?

This might be good enough for a first version but we'll need to closely measure the impact on CPU time consumed by this kind of worker with something like `stackprof`. It's difficult to tell now because there are many factors that can have an impact like number of this kind of jobs generated, the average `from`-`to` interval, etc.

At least we should add a TODO here explaining what this does and mention that this is probably something that could be improved.

Reply:
isn't just a subset of the keys generated/removed (length) ?

Reply:
it is generating all keys up to `offset + length`.

I will write an issue about improving the algorithm to use indexes. 

Deleting service stats is not a job that will be created often

Reply:
Only a subset is deleted, but all of them are generated `stats_key_gen.keys`.

Reply:
:+1: , I was thinking on the old implementation, when enumerators were used and not all keys were generated.